src/IsaacApiTypes.tsx

```diff
@@ -1,724 +1,13 @@
-  quizId?: string;
-  quizSummary?: ContentSummaryDTO;
-}
-
-export interface Mark {
-  correct?: number;
-  incorrect?: number;
-  notAttempted?: number;
-}
-
-export interface ExternalReference {
-  title?: string;
-  url?: string;
-}
-
-export interface ContentBase {
-  id?: string;
-  type?: string;
-  tags?: string[];
-  canonicalSourceFile?: string;
-  version?: string;
-  audience?: AudienceContext[];
-  display?: { [index: string]: string[] };
-}
-
-export interface Content extends ContentBase {
-  title?: string;
-  subtitle?: string;
-  author?: string;
-  encoding?: string;
-  layout?: string;
-  children?: ContentBase[];
-  value?: string;
-  attribution?: string;
-  relatedContent?: string[];
-  published?: boolean;
-  level?: number;
-  searchableContent?: string;
-}
-
-export interface ResultsWrapper<T> {
-  results?: T[];
-  totalResults?: number;
-}
-
-export interface Address {
-  addressLine1?: string;
-  addressLine2?: string;
-  town?: string;
-  county?: string;
-  postalCode?: string;
-  country?: string;
-}
-
-export interface AnsweredQuestionsByDate {
-  [date: string]: number;
-}
-
-export interface TOTPSharedSecretDTO {
-  userId: number;
-  sharedSecret: string;
-  created: EpochTimeStamp;
-}
-
-export interface MisuseStatisticDTO {
-  agentIdentifier: string;
-  eventType: string;
-  isMisused: boolean;
-  isOverSoftThreshold: boolean;
-  lastEventTimestamp?: number;
-  currentCounter: number;
-}
-
-export type GameboardCreationMethod = "FILTER" | "BUILDER";
-
-export type EventStatus = "OPEN" | "FULLY_BOOKED" | "CANCELLED" | "CLOSED" | "WAITING_LIST_ONLY";
-
-export type FastTrackConceptState = "ft_top_ten" | "ft_upper" | "ft_lower";
-
-export type BookingStatus = "CONFIRMED" | "CANCELLED" | "WAITING_LIST" | "ATTENDED" | "ABSENT" | "RESERVED";
-
-export const USER_ROLES = [
-  "STUDENT",
-  "TUTOR",
-  "TEACHER",
-  "EVENT_LEADER",
-  "CONTENT_EDITOR",
-  "EVENT_MANAGER",
-  "ADMIN",
-] as const;
-export type UserRole = (typeof USER_ROLES)[number];
-
-export type EmailVerificationStatus = "VERIFIED" | "NOT_VERIFIED" | "DELIVERY_FAILED";
-
-export type QuizFeedbackMode = "NONE" | "OVERALL_MARK" | "SECTION_MARKS" | "DETAILED_FEEDBACK";
-
-export type GroupMembershipStatus = "ACTIVE" | "INACTIVE" | "DELETED";
-
-export type Gender = "MALE" | "FEMALE" | "OTHER" | "PREFER_NOT_TO_SAY" | "UNKNOWN";
-
-export type AuthenticationProvider = "GOOGLE" | "FACEBOOK" | "TWITTER" | "RAVEN" | "TEST" | "SEGUE";
-
-export type GameboardItemState = "PERFECT" | "PASSED" | "IN_PROGRESS" | "NOT_ATTEMPTED" | "FAILED";
-
-export type QuestionPartState = "CORRECT" | "INCORRECT" | "NOT_ATTEMPTED";
+export * from "./IsaacApiTypesGenerated";
+export * from "./IsaacApiTypesCustom";
+
+// This file imports all the types auto generated by maven typescript-generator plugin in API
+// It then imports any custom types, and re-exports any that we need to override or extend
+// This allows us to use the auto-generated file without editing it
```

Comment:
I like this comment, and I think we should probably include a warning saying that the `IsaacApiTypesAutoGenerated` is not meant to be changed, and that any changes need to happen in `IsaacApiTypesCustom`. You have already alluded to this, but the more obvious and clear it is, the better. This is to avoid anyone else in the future making the same mistake again.